integration-cli/docker_api_containers_test.go

```diff
@@ -24,31 +25,24 @@ import (
 func (s *DockerSuite) TestContainerApiGetAll(c *check.C) {
 	testRequires(c, DaemonIsLinux)
 	startCount, err := getContainerCount()
-	if err != nil {
-		c.Fatalf("Cannot query container count: %v", err)
-	}
+	c.Assert(err,checker.IsNil,Commentf("Cannot query container count: %v", err))
 
 	name := "getall"
 	dockerCmd(c, "run", "--name", name, "busybox", "true")
 
 	status, body, err := sockRequest("GET", "/containers/json?all=1", nil)
-	c.Assert(err, check.IsNil)
-	c.Assert(status, check.Equals, http.StatusOK)
+	c.Assert(err, checker.IsNil)
+	c.Assert(status, checker.Equals, http.StatusOK)
 
 	var inspectJSON []struct {
 		Names []string
 	}
-	if err = json.Unmarshal(body, &inspectJSON); err != nil {
-		c.Fatalf("unable to unmarshal response body: %v", err)
-	}
 
-	if len(inspectJSON) != startCount+1 {
-		c.Fatalf("Expected %d container(s), %d found (started with: %d)", startCount+1, len(inspectJSON), startCount)
-	}
+	c.Assert(json.Unmarshal(body, &inspectJSON),checker.IsNil,Commentf("unable to unmarshal response body: %v", json.Unmarshal(body, &inspectJSON)))
```

Comment:
This ain't gonna work, you call twice `json.Unmarshal` but I don't even think you need a comment here.